src/main/scala/com/tuplejump/kafka/connect/cassandra/CassandraSinkTask.scala

```diff
@@ -51,33 +55,107 @@ class CassandraSinkTask extends SinkTask with TaskLifecycle {
 
 /** INTERNAL API. */
 private[kafka] object CassandraSinkTask {
+
   import Configuration._
 
+  import scala.util.parsing.json._
+
   /* TODO: Use keySchema, partition and kafkaOffset
    TODO: Add which types are currently supported in README */
-  def convert(record: SinkRecord, sink: SinkConfig): Query = {
+  def convert(record: SinkRecord, sink: SinkConfig, fieldMappingProperty: Option[String] = None): Query = {
+    val colNamesVsValues: Map[String, String] = {
+      fieldMappingProperty match {
+        case Some(fieldMappingString) => convertToCqlData(record, fieldMappingString)
+        case None => convertToCqlData(record)
+      }
+    }
+    colNamesVsValues.view.map(e => Vector(e._1, e._2)).transpose match {
+      case columnNames :: columnValues :: Nil =>
+        s"INSERT INTO ${sink.namespace}(${columnNames.mkString(",")}) VALUES(${columnValues.mkString(",")})"
+    }
+  }
+
+  def convertToCqlData(record: SinkRecord): (Map[String, String]) = {
     val valueSchema = record.valueSchema
-    val columnNames = valueSchema.fields.asScala.map(_.name).toSet
-    val columnValues = valueSchema.`type`() match {
+    valueSchema.`type`() match {
       case STRUCT =>
         val struct: Struct = record.value.asInstanceOf[Struct]
-        columnNames.map(schema(valueSchema, struct, _)).mkString(",")
+        valueSchema.fields.asScala.map { field =>
+          (field.name, schema(valueSchema, struct, field))
+        }.toMap
       case other =>
         throw new DataException(
           s"Unable to create insert statement with unsupported value schema type $other.")
     }
-    s"INSERT INTO ${sink.namespace}(${columnNames.mkString(",")}) VALUES($columnValues)"
   }
 
   /* TODO support all types. */
-  def schema(valueSchema: Schema, result: Struct, col: String): AnyRef =
-    valueSchema.field(col).schema match {
+  def schema(valueSchema: Schema, result: Struct, field: Field): String = {
+    field.schema match {
       case x if x.`type`() == Schema.STRING_SCHEMA.`type`() =>
-        s"'${result.get(col).toString}'"
+        s"'${result.get(field).toString}'"
       case x if x.name() == Timestamp.LOGICAL_NAME =>
-        val time = Timestamp.fromLogical(x, result.get(col).asInstanceOf[JDate])
+        val time = Timestamp.fromLogical(x, result.get(field).asInstanceOf[JDate])
         s"$time"
       case y =>
-        result.get(col)
+        String.valueOf(result.get(field))
+    }
+  }
+
+
+  // scalastyle:off
+  def convertToCqlData(record: SinkRecord, fieldMappingString: String): Map[String, String] = {
```

Comment:
We could use a `Map[String,String]` instead of `fieldMappingString`. Parsing String to get a JSON object may not be the ooptimal solution.